extra/src/main/java/aima/extra/probability/DoubleProbabilityNumber.java

```diff
@@ -72,70 +65,31 @@ public DoubleProbabilityNumber(double value) {
 	 */
 	public DoubleProbabilityNumber(BigDecimal value) {
 		if (null == value || value.compareTo(new BigDecimal(0)) == -1 || value.compareTo(new BigDecimal(1)) == 1) {
```

Comment:
if (null == value || value.compareTo(BigDecimal.ZERO) == -1 || value.compareTo(BigDecimal.ONE) == 1) {

is more efficient/cleaner.